core/range/minmax_spec.rb

```diff
@@ -0,0 +1,66 @@
+require_relative '../../spec_helper'
+
+describe 'Range#minmax' do
+  before(:each) do
+    @x = mock('x')
+    @y = mock('y')
+
+    @x.should_receive(:<=>).with(@y).any_number_of_times.and_return(-1)
+    @x.should_receive(:<=>).with(@x).any_number_of_times.and_return(0)
+    @y.should_receive(:<=>).with(@x).any_number_of_times.and_return(1)
+    @y.should_receive(:<=>).with(@y).any_number_of_times.and_return(0)
+  end
+
+  ruby_version_is '2.7' do # TODO: Remove this before merging!
+    class Range
+      # This monkey patch roughly matches the implementation of the bug fix in MRI's range.c
+      def minmax
+        return super if block_given?
+
+        [min, max]
+      end
+    end
+  end
+
+  describe 'by enumerating the range' do
+    before(:each) do
+      @x.should_receive(:succ).once.and_return(@y)
+    end
+
+    ruby_version_is ''...'2.7' do
+      it 'should return the minimum and maximum elements' do
+        (@x..@y).minmax.should == [@x, @y]
+      end
+    end
+
+    it 'should not return the end of the range as maximum if exclusive range' do
+      (@x...@y).minmax.should == [@x, @x]
+    end
+
+    it 'should respect custom comparison block' do
+      (@x..@y).minmax { |x, y| - (x <=> y) }.should == [@y, @x]
+    end
+  end
+
+  describe 'without enumerating the range' do
+    before(:each) do
+      # Prevent enumeration, but allow members to respond_to? :succ
+      @x.should_not_receive(:succ)
+      @y.should_not_receive(:succ)
+    end
+
+    ruby_version_is '2.7' do
+      it 'should return the minimum and maximum elements' do
+        (@x..@y).minmax.should == [@x, @y]
+      end
```

Comment:
What happens if we do the same with an exclusive Range? I think we should spec that (even if it's an exception)

Reply:
I can add a spec for that in the section that permits enumerating the range.

Reply:
Actually `it 'should not return the end of the range as maximum if exclusive range'` effectively already tests that, but maybe it could do with a better naming.

Reply:
I see, I missed that, it's fine then.